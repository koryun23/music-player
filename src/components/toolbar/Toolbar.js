import React from "react";
import "../../css/toolbar/Toolbar.css";
import PlayAll from "./PlayAll";
import AddAll from "./AddAll";
import Filter from "./Filter";
import { Container, Nav, NavDropdown, Navbar, Form, Button, Col, Row } from "react-bootstrap";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faSearch } from "@fortawesome/free-solid-svg-icons";

export default function Toolbar(props) {
    return (
        <div className="container-fluid">
            <div className="toolbar row">
                <div className="col-md-2 col-sm-4 col-xs-12 custom-column">
                    <PlayAll onPlayAll={props.onPlayAll}/>
                </div>
                <div className="col-md-2 col-sm-4 col-xs-12 custom-column">
                    <AddAll onAddAll={props.onAddAll}/>
                </div>
                <div className="col-md-6 col-sm-1 custom-column"></div>
                <div className="col-md-2 col-sm-3 col-xs-12 custom-column">
                    <Filter onFilter={props.onFilter}/>
                </div>
                <div className="col-sm-1"></div>
            </div>
        </div>

    );
}