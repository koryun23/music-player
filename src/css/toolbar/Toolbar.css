.toolbar{
    background-color: rgb(229, 229, 229);
    padding: 10px;
    border-radius: 10px;
    height: fit-content;
}
.play-all-button {
    border-radius: 10px;
    border: 1px solid rgb(185, 185, 185);
    height: fit-content;
    width: fit-content;
}

.add-all-button {
    border-radius: 10px;
    border: 1px solid rgb(185, 185, 185);
    height: fit-content;
    width: fit-content;
}

.filter-input-field {
    border-radius: 20px;
    border: 1px solid rgb(185, 185, 185);
    height: auto;
    padding: 5px;
    width: 100%;
}

.play-icon {
    display: inline-block;
    margin-right: 5px;
}
.add-icon {
    display: inline-block;
    margin-right: 10px;
}
.button-description{
    display: inline-block;
    padding: auto;
}
.options {
    display: inline-block;
    height: 100%;
    width: 20%;
    position: absolute;
    right: 0;
    top: 0;
    bottom: 0;
    border-top-right-radius: 7px;
    border-bottom-right-radius: 7px;
    border-right: 0;
    border-top: 0;
    border-bottom: 0;
}
.down-arrow-icon {
    margin-right: 10px;
    display: inline-block;
}
.up-arrow-icon {
    display: inline-block;
}

.filter-form {
    height: fit-content;
    width: fit-content;
}

.custom-column {
    height: 100%;
    margin:auto;
}