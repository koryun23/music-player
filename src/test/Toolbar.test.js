import renderer from "react-test-renderer";
import Toolbar from "../components/toolbar/Toolbar";

it('renders correctly', () => {
    const tree = renderer.create(<Toolbar />).toJSON();
    expect(tree).toMatchSnapshot();
})