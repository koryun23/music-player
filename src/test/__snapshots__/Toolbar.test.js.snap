// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container-fluid"
>
  <div
    className="toolbar row"
  >
    <div
      className="col-md-2 col-sm-4 col-xs-12 custom-column"
    >
      <div
        className="dropdown dropdown-div play-all-button"
        onClick={[Function]}
      >
        <button
          aria-expanded="false"
          aria-haspopup="true"
          className="button-description btn btn-light dropdown-toggle"
          data-toggle="dropdown"
          id="dropdownMenuButton"
          type="button"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-play fa-lg play-icon"
            data-icon="play"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M73 39c-14.8-9.1-33.4-9.4-48.5-.9S0 62.6 0 80V432c0 17.4 9.4 33.4 24.5 41.9s33.7 8.1 48.5-.9L361 297c14.3-8.7 23-24.2 23-41s-8.7-32.2-23-41L73 39z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <b>
            Play All
          </b>
        </button>
        <div
          aria-labelledby="dropdownMenuButton"
          className="dropdown-menu"
        >
          <a
            className="dropdown-item"
            href="/"
            onClick={[Function]}
          >
            Repeat After Queue Ends
          </a>
          <a
            className="dropdown-item"
            href="/"
            onClick={[Function]}
          >
            Play Queue Only Once
          </a>
        </div>
      </div>
    </div>
    <div
      className="col-md-2 col-sm-4 col-xs-12 custom-column"
    >
      <div
        className="add-all-button dropdown dropdown-div"
        onClick={[Function]}
      >
        <button
          aria-expanded="false"
          aria-haspopup="true"
          className="button-description btn btn-light dropdown-toggle"
          data-toggle="dropdown"
          id="dropdownMenuButton"
          type="button"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-plus fa-lg add-icon"
            data-icon="plus"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 80c0-17.7-14.3-32-32-32s-32 14.3-32 32V224H48c-17.7 0-32 14.3-32 32s14.3 32 32 32H192V432c0 17.7 14.3 32 32 32s32-14.3 32-32V288H400c17.7 0 32-14.3 32-32s-14.3-32-32-32H256V80z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <b>
            Add All
          </b>
        </button>
        <div
          className="dropdown-menu"
        >
          <a
            className="dropdown-item"
            href="/"
            onClick={[Function]}
          >
            Add All Sequentially
          </a>
          <a
            className="dropdown-item"
            href="/"
            onClick={[Function]}
          >
            Add All Shuffled
          </a>
        </div>
      </div>
    </div>
    <div
      className="col-md-6 col-sm-1 custom-column"
    />
    <div
      className="col-md-2 col-sm-3 col-xs-12 custom-column"
    >
      <form>
        <div
          className="filter-form"
        >
          <input
            className="filter-input-field"
            onChange={[Function]}
            placeholder="ï€‚ Filter"
            style={
              Object {
                "fontFamily": "Arial, FontAwesome",
              }
            }
            type="text"
            value=""
          />
        </div>
      </form>
    </div>
    <div
      className="col-sm-1"
    />
  </div>
</div>
`;
